src/test/java/org/folio/circulation/domain/OverduePeriodCalculatorServiceTest.java

```diff
@@ -0,0 +1,200 @@
+package org.folio.circulation.domain;
+
+import api.support.builders.LoanBuilder;
+import api.support.builders.LoanPolicyBuilder;
+import api.support.builders.OverdueFinePolicyBuilder;
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.apache.commons.lang3.ObjectUtils;
+import org.folio.circulation.domain.policy.LoanPolicy;
+import org.folio.circulation.domain.policy.OverdueFinePolicy;
+import org.folio.circulation.domain.policy.Period;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.LocalDate;
+import org.joda.time.LocalTime;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import static api.support.fixtures.OpeningHourExamples.afternoon;
+import static api.support.fixtures.OpeningHourExamples.allDay;
+import static api.support.fixtures.OpeningHourExamples.morning;
+import static org.folio.circulation.domain.OpeningDay.createOpeningDay;
+import static org.joda.time.DateTimeConstants.MINUTES_PER_DAY;
+import static org.joda.time.DateTimeConstants.MINUTES_PER_HOUR;
+import static org.joda.time.DateTimeConstants.MINUTES_PER_WEEK;
+import static org.junit.Assert.*;
+
+@RunWith(JUnitParamsRunner.class)
+public class OverduePeriodCalculatorServiceTest {
```

Comment:
I think it would be good if these tests could cover all of the code within the service. If that requires mocking the calendar repository (by renaming it, extracting an interface and mocking the interface) I think that is ok / preferable.

It's ok if we want to defer that work to the follow up story that starts to integrate this work into the behaviour of the system.

Reply:
> It's ok if we want to defer that work to the follow up story that starts to integrate this work into the behaviour of the system.

That is exactly the plan.

Reply:
Ok, in which case I think we should remove any code not needed for this standalone change until it is needed for the  integration work

Reply:
I believe this is already done.

Reply:
SonarQube is reporting warnings and coverage that suggests that isn't the case.

Reply:
I think it might well be time to accept that some of the code won't be used until the integration work is done.

Please can the warning for unused code be resolved before this merged.

Reply:
done